HelloSignalR/Startup.cs

```diff
@@ -0,0 +1,134 @@
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddAuthentication();
+            services.AddSignalR();
+            services.AddCaching();
+        }
+
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory, ILogger<Startup> logger)
+        {
+            // Add the platform handler to the request pipeline.
+            app.UseIISPlatformHandler();
+
+            if (env.IsDevelopment())
+            {
+                // In case any errors occur.
+                app.UseDeveloperExceptionPage();
+
+                if (env.IsDevelopment())
+                {
+                    loggerFactory.MinimumLevel = LogLevel.Information;
+                    loggerFactory.AddConsole();
+                    loggerFactory.AddDebug();
+                }
+            }
+
+            logger.LogInformation($"Environment: {env.EnvironmentName}");
+
+            var hostname = "http://localhost:5000/";
+
+            // Add token issuing middleware.
+            app.UseOpenIdConnectServer(config =>
+            {
+                // Your Auhentication server.
+                config.Provider = new AuthenticationProvider();
+                config.Issuer = new Uri(hostname);
+
+                if (env.IsDevelopment())
+                {
+                    // For not requiring HTTPS in localhost.
+                    config.AllowInsecureHttp = true;
+                    config.ApplicationCanDisplayErrors = true;
+                }
+            });
+
+            app.UseJwtBearerAuthentication(options =>
+            {
+                // We need this to enable authentication for all requests.
+                options.AutomaticAuthenticate = true;
+
+                // Your audience that has to be allowed by your Authority server.
+                options.Audience = hostname;
+
+                // Authority that issued your token (i.e. your identity server).
+                options.Authority = hostname;
+
+                options.Events = new JwtBearerEvents
+                {
+                    // For SignalR connections default Authorization header does not work,
+                    // Because there are no Headers in WebSockets specification
+                    // This is why we have to implement our own way to retrieve token
+                    // Which can be as easy as adding it to query string
+                    OnReceivingToken = context =>
+                    {
+                        string token = null;
+                        // TODO: Maybe these constants are defined somewhere in framework already?
+                        const string AuthorizationKey = "Authorization";
+                        const string BearerKey = "Bearer";
+
+                        // First, try to extract token from 'Authorization: Bearer {token}' header
+                        if (context.Request.Headers.ContainsKey(AuthorizationKey))
```

Comment:
The JWT bearer middleware already does that for you if you leave `context.Token` to null. Consider removing this whole block :smile: